### PR TITLE
[production/RRFS.v1] Use production utility cpreq, other miscellaneous fixes

### PR DESCRIPTION
**Pull request by BenjaminBlake-NOAA**
## DESCRIPTION OF CHANGES: 

- This PR includes several components which address some of the current open issues.
- The `cp` command is replaced with `cpreq -p` in the ex-scripts (issue #407 ) for instances where input files needed for running the task are retrieved.
- The `wait` command was removed from exrrfs_prdgen.sh (issue #406 ).  The section of code containing this command is not normally executed anyway for RRFS anyway.
- The two instances of `rm -f fort.*` have been removed from exrrfs_bufrsnd.sh and exrrfs_post.sh (issue #404 ).  The `prep_step` production utility is called further down in each script.
- `$PARMfv3` was changed to `$PARMrrfs` within exrrfs_bufrsnd.sh.  This path should eventually be set within the J-job and not in the ex-script.
- A bug was discovered related to the `$GESROOT` path definition.  The default value of `$GESROOT` is now set to `$PTMP/nwges` within setup.sh.  This is similar to what is done for `$COMROOT`, which is set to `$PTMP/com`.

## TESTS CONDUCTED: 

Successful tests were completed with the RRFS fire weather nest and DA engineering test spinup cycle.

### Machines/Platforms:

- WCOSS2
  - [x] Cactus/Dogwood
  - [ ] Acorn
- RDHPCS
  - [ ] Hera
  - [ ] Jet
  - [ ] Orion
  - [ ] Hercules

### Test cases: 

- [ ] Engineering tests
  - [ ] Non-DA engineering test
  - [x] DA engineering test
    - [ ] Retro
    - [ ] Ensemble
    - [x] Parallel
- [x] RRFS fire weather
- [ ] RRFS_A:
- [ ] RRFS_B:
- [ ] RTMA:
- [ ] Others:

## ISSUE: 

- Fixes issues #407 , #406 , and #404

### CODE REVIEW

**Review by MatthewPyle-NOAA — COMMENTED** [review]
I added one comment, and wanted to start a conversation about it.  I don't think we want to do it as so much of a universal change.  The intent of cpreq is to make sure that something needed is copied to prevent a weird executable failure (think of a model not copying in an input file).  I might want to go through more carefully and think about which ones maybe don't need to be changed to cpreq.  

Also, was looking at the HAFS system to see how they deal with something like cpreq for a system that runs on non-WCOSS machines.  They use a variable that is defined as cpreq if available, and a regular /bin/cp if not.  Not advocating that kind of change here right now, but something to think about in the long run.

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA Thanks for your feedback.  I was also thinking cpreq might only be needed in certain situations and not as a universal change, but I wasn't entirely sure so I wanted to see what you thought.  I'll go through and revert some of these cpreq instances back to cp, then maybe you can take another look at this PR.  I'll convert this PR to a draft for now.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Looks good, with one small suggestion.

File: scripts/exrrfs_bufrsnd.sh
[the remark this answers is not in the document]
Comment (BenjaminBlake-NOAA) [reply scripts/exrrfs_bufrsnd.sh:1]: @MatthewPyle-NOAA I agree, and sure I'll remove that.  I see another hardwired path in exrrfs_bufrsnd.sh (GEMPAKrrfs) which should probably be defined at the J-job level.  In an upcoming PR I can add the standard HOMErrfs, EXECrrfs, SCRIPTSrrfs, USHrrfs, etc. path definitions to the J-jobs.

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA I am going to go ahead and merge this PR since you approved the changes on Friday.  Thanks

**Pull request merged**